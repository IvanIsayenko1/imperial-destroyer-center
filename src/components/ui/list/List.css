.list {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  gap: 1.5rem;
}

@keyframes fadeIn {
  from {
    opacity: 0;
    transform: translateY(10px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}

@media (max-width: 640px) {
  .list {
    grid-template-columns: 1fr;
  }
}
