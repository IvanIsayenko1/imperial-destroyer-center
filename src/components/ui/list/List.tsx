import React from "react";
import "./List.css";

interface ListProps<T> {
  items: T[];
  renderItem: (item: T) => React.ReactNode;
}

export const List = <T,>({ items, renderItem }: ListProps<T>) => {
  return (
    <div className="list">
      {items.map((item, index) => (
        <div key={index}>{renderItem(item)}</div>
      ))}
    </div>
  );
};
