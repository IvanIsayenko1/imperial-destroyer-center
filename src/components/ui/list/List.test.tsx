import { render, screen } from "@testing-library/react";
import { List } from "./List";

describe("List component", () => {
  it("should render list items correctly", () => {
    const items = [1, 2, 3];
    const renderItem = (item: number) => <span>{item}</span>;

    render(<List items={items} renderItem={renderItem} />);

    items.forEach((item) => {
      expect(screen.getByText(item.toString())).toBeInTheDocument();
    });
  });

  it("should render no items when list is empty", () => {
    const renderItem = (item: number) => <span>{item}</span>;

    render(<List items={[]} renderItem={renderItem} />);

    // Expect not to find any item in the document
    expect(screen.queryByText("1")).not.toBeInTheDocument();
  });

  it("should render custom item content using renderItem function", () => {
    const items = ["apple", "banana", "cherry"];
    const renderItem = (item: string) => <strong>{item}</strong>;

    render(<List items={items} renderItem={renderItem} />);

    items.forEach((item) => {
      expect(screen.getByText(item)).toBeInTheDocument();
      expect(screen.getByText(item).tagName).toBe("STRONG");
    });
  });

  it("should match snapshot", () => {
    const items = [1, 2, 3];
    const renderItem = (item: number) => <span>{item}</span>;

    const { asFragment } = render(
      <List items={items} renderItem={renderItem} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
