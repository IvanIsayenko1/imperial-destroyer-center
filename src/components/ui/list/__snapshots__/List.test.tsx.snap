// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`List component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="list"
  >
    <div>
      <span>
        1
      </span>
    </div>
    <div>
      <span>
        2
      </span>
    </div>
    <div>
      <span>
        3
      </span>
    </div>
  </div>
</DocumentFragment>
`;
